x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -290,36 +242,47 @@ export const EditConnector = React.memo(
             <EuiFlexItem grow={false} data-test-subj="connector-edit-header">
               <h4>{i18n.CONNECTORS}</h4>
             </EuiFlexItem>
-            {isLoading && <EuiLoadingSpinner data-test-subj="connector-loading" />}
-            {!isLoading && !editConnector && permissions.push && actionsReadCapabilities && (
```

Comment:
I think it is better to not show the loading and disable the pencil button when the logic condition is satisfied.